common/src/main/java/org/vivecraft/client_vr/provider/openxr/MCOpenXR.java

```diff
@@ -0,0 +1,1494 @@
+            grip_left.next(NULL);
+            grip_left.action(new XrAction(grip[0], actionSet));
+            grip_left.subactionPath(getPath("/user/hand/right"));
+            grip_left.poseInActionSpace(POSE_IDENTITY);
+            PointerBuffer pp = stackCallocPointer(1);
+            error = XR10.xrCreateActionSpace(session, grip_left, pp);
+            logError(error, "xrCreateActionSpace",  "grip: /user/hand/right");
+            this.gripSpace[0] = new XrSpace(pp.get(0), session);
+
+            grip_left.action(new XrAction(grip[1], actionSet));
+            grip_left.subactionPath(getPath("/user/hand/left"));
+            error = XR10.xrCreateActionSpace(session, grip_left, pp);
+            logError(error, "xrCreateActionSpace",  "grip: /user/hand/left");
+            this.gripSpace[1] = new XrSpace(pp.get(0), session);
+
+            grip_left.action(new XrAction(aim[0], actionSet));
+            grip_left.subactionPath(getPath("/user/hand/right"));
+            error = XR10.xrCreateActionSpace(session, grip_left, pp);
+            logError(error, "xrCreateActionSpace",  "aim: /user/hand/right");
+            this.aimSpace[0] = new XrSpace(pp.get(0), session);
+
+            grip_left.action(new XrAction(aim[1], actionSet));
+            grip_left.subactionPath(getPath("/user/hand/left"));
+            error = XR10.xrCreateActionSpace(session, grip_left, pp);
+            logError(error, "xrCreateActionSpace",  "aim: /user/hand/left");
+            this.aimSpace[1] = new XrSpace(pp.get(0), session);
+
+        }
+    }
+
+    public long getPath(String pathString) {
+        return this.paths.computeIfAbsent(pathString, s -> {
+            try (MemoryStack ignored = stackPush()) {
+                LongBuffer buf = stackCallocLong(1);
+                int error = XR10.xrStringToPath(instance, pathString, buf);
+                logError(error, "getPath",  pathString);
+                return buf.get();
+            }
+        });
+    }
+
+    private long createAction(String name, String localisedName, String type, XrActionSet actionSet) {
+        try (MemoryStack stack = MemoryStack.stackPush()){
+            String s = name.split("/")[name.split("/").length -1].toLowerCase();
+            XrActionCreateInfo hands = XrActionCreateInfo.calloc(stack);
+            hands.type(XR10.XR_TYPE_ACTION_CREATE_INFO);
+            hands.next(NULL);
+            hands.actionName(memUTF8(s));
+            switch (type) {
+                case "boolean" -> hands.actionType(XR10.XR_ACTION_TYPE_BOOLEAN_INPUT);
+                case "vector1" -> hands.actionType(XR10.XR_ACTION_TYPE_FLOAT_INPUT);
+                case "vector2" -> hands.actionType(XR10.XR_ACTION_TYPE_VECTOR2F_INPUT);
+                case "pose" -> hands.actionType(XR10.XR_ACTION_TYPE_POSE_INPUT);
+                case "haptic" -> hands.actionType(XR10.XR_ACTION_TYPE_VIBRATION_OUTPUT);
+            }
+            hands.countSubactionPaths(0);
+            hands.subactionPaths(null);
```

Comment:
Replace line 1358 with the following
`if(subactionPaths != null) {
                LongBuffer buffer = stackCallocLong(subactionPaths.length);
                for(String path : subactionPaths) {
                    buffer.put(getPath(path));
                }
                hands.countSubactionPaths(subactionPaths.length);
                hands.subactionPaths(buffer.rewind());
            } else {
                hands.countSubactionPaths(0);
                hands.subactionPaths(null);
            }`

OpenXR spec states that XR_ERROR_PATH_UNSUPPORTED **must** be returned if, during the xrCreateActionSpace call, the subactionPaths values do not contain the value of subactionPath. So line 1307 fails on Oculus/Meta XR Runtime

Reply:
(Make sure to also add the parameter String[] subactionPaths to the createAction method, it can be null for the call at 1210 for ex, it won't break)